mtags/src/main/scala-3/scala/meta/internal/pc/CompletionProvider.scala

```diff
@@ -110,30 +110,32 @@ class CompletionProvider(
 
   extension (l: List[CompletionValue])
     def filterInteresting(): (List[CompletionValue], SymbolSearch.Result) =
+
       val isSeen = mutable.Set.empty[String]
       val buf = List.newBuilder[CompletionValue]
       def visit(head: CompletionValue): Boolean =
-        val sym = head.symbol
-        val id = head.kind match
-          case CompletionValue.Kind.NamedArg =>
-            sym.detailString + "="
-          case CompletionValue.Kind.Keyword =>
-            head.label
-          case _ =>
-            val name = SemanticdbSymbols.symbolName(sym)
-            if sym.isClass || sym.is(Module) then
-              // drop #|. at the end to avoid duplication
-              name.substring(0, name.length - 1)
-            else name
-        def isNotLocalForwardReference: Boolean =
-          !sym.isLocalToBlock ||
-            !sym.srcPos.isAfter(pos) ||
-            sym.is(Param)
-
-        if !isSeen(id) &&
-          !isUninterestingSymbol(sym) &&
-          isNotLocalForwardReference
-        then
+        val (id, include) =
+          head match
+            case symOnly: CompletionValue.SymbolOnly =>
+              val sym = symOnly.symbol
+              val name = SemanticdbSymbols.symbolName(sym)
+              val id =
+                if sym.isClass || sym.is(Module) then
+                  // drop #|. at the end to avoid duplication
+                  name.substring(0, name.length - 1)
+                else name
+
+              val include =
+                !isUninterestingSymbol(sym) &&
```

Comment:
Gentle off top, I wonder why `isUninterestingSymbol` and `isEvilMethod` are lazy vals, is there a reason for that?

Reply:
I think there might be a point in `lazy` there but it's hard to say how much time they save.
Hypothetically, if there is no symbol completions then these set won't be required for filtering.